core/src/main/java/org/infinispan/persistence/sifs/IndexNode.java

```diff
@@ -675,6 +675,9 @@ private IndexNode copyWith(IndexRequest request, int cacheSegment, byte[] indexK
       if (leafNodes.length == 0) {
          overwriteHook.setOverwritten(request, cacheSegment, false, -1, -1);
          if (overwriteHook.check(request, -1, -1)) {
+            if (overwriteHook.isRemoval(request)) {
+               segment.getCompactor().free(file, size);
```

Comment:
IIUC, the compactor waits for sent index updates to complete. Effectively, does this only update the file stats?

I ask because it could put more pressure on the compactor(?). If that's even an issue.

Reply:
This updates the file stats which affects when the compactor runs. This line is only hit when you have a remove that is fired when the key is not in SIFS, so that data is already effectively free since the REMOVE didn't write any new value.

There is no additional pressure on the compactor, this just makes sure the stats are correct in the first place.

Without this the fileStats will say this file has valid bytes for a REMOVE, which is not what we want.